packages/auto-approve/src/auto-approve.ts

```diff
@@ -222,8 +229,12 @@ export function handler(app: Probot) {
         return;
       }
 
+      const octokit = await getAuthenticatedOctokit(
+        context.payload.installation!.id
```

Comment:
I think this will fail when installation ID doesn't exist, which we don't want to have [happen](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator).  I think we want to attempt to authenticate only if the installation ID exists.

Reply:
It's required for compilation because the schema (or interface) claims it's an optional field, but there should always be the installation id in webhooks that the bot receives.

Without attempting to authenticate, we cannot compile `getChangedFiles` because `context.octokit` is the `Octokit` from v18, but we need `Octokit` from v19.

What do you suggest otherwise?

Reply:
We should probably add a branch like:

```
let octokit;
if (installation.id) {
      octokit = await getAuthenticatedOctokit(
        context.payload.installation!.id
       )
} else {
    throw new Error('Installation ID not provided in ${context.payload.event}; cannot authenticate for getting files`)
}
```

Reply:
It will likely be dead code, but somehow installation id is missing it will give a better error message.

Reply:
Done!